pkg/machine/hyperv/vsock.go

```diff
@@ -255,16 +256,16 @@ func (hv *HVSockRegistryEntry) Listen() error {
 			logrus.Error(err)
 		}
 	}()
-	conn, err := listener.Accept()
-	if err != nil {
-		return err
-	}
+
+	errChan := make(chan error)
+	connChan := make(chan net.Conn)
+	go machine.ListenAndWaitOnSocket(errChan, connChan, listener)
+	conn := <-connChan
+
 	defer func() {
 		if err := conn.Close(); err != nil {
```

Comment:
conn can be nil, so this close might cause a nil pointer panic. You should either check the error before the close or check for != nil on the conn.

Reply:
Done.